manager/src/test/java/com/redhat/service/bridge/manager/workers/resources/ConnectorWorkerTest.java

```diff
@@ -1,84 +1,90 @@
 import com.redhat.service.bridge.manager.connectors.ConnectorsApiClient;
+import com.redhat.service.bridge.manager.dao.BridgeDAO;
 import com.redhat.service.bridge.manager.dao.ConnectorsDAO;
+import com.redhat.service.bridge.manager.dao.ProcessorDAO;
+import com.redhat.service.bridge.manager.models.Bridge;
 import com.redhat.service.bridge.manager.models.ConnectorEntity;
+import com.redhat.service.bridge.manager.models.Processor;
 import com.redhat.service.bridge.manager.models.Work;
+import com.redhat.service.bridge.manager.utils.DatabaseManagerUtils;
+import com.redhat.service.bridge.manager.utils.Fixtures;
 import com.redhat.service.bridge.manager.workers.WorkManager;
 import com.redhat.service.bridge.rhoas.RhoasTopicAccessType;
+import com.redhat.service.bridge.test.resource.PostgresResource;
+
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.mockito.InjectMock;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatCode;
-import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.Mockito.atLeastOnce;
 import static org.mockito.Mockito.atMostOnce;
 import static org.mockito.Mockito.never;
-import static org.mockito.Mockito.spy;
 import static org.mockito.Mockito.times;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
-@ExtendWith(MockitoExtension.class)
+@QuarkusTest
+@QuarkusTestResource(PostgresResource.class)
 class ConnectorWorkerTest {
 
+    private static final String TEST_PROCESSOR_NAME = "TestProcessor";
+    private static final String TEST_CONNECTOR_NAME = "TestConnector";
     private static final String TEST_CONNECTOR_EXTERNAL_ID = "connectorExternalId";
     private static final String TEST_RESOURCE_ID = "123";
-    private static final String TEST_TOPIC_NAME = "topicName";
-
-    @Mock
-    ConnectorsDAO connectorsDAO;
+    private static final String TEST_TOPIC_NAME = "TopicName";
 
-    @Mock
+    @InjectMock
     RhoasService rhoasService;
 
-    @Mock
+    @InjectMock
     ConnectorsApiClient connectorsApi;
 
-    @Mock
+    @Inject
     WorkManager workManager;
 
-    @Mock
-    EntityManager entityManager;
+    @Inject
+    ConnectorWorker worker;
 
-    private ConnectorWorker worker;
+    @Inject
+    BridgeDAO bridgeDAO;
+
+    @Inject
+    ProcessorDAO processorDAO;
+
+    @Inject
+    ConnectorsDAO connectorsDAO;
+
+    @Inject
+    DatabaseManagerUtils databaseManagerUtils;
 
     @BeforeEach
-    void setup() {
-        this.worker = new ConnectorWorker();
-        this.worker.connectorsDAO = this.connectorsDAO;
-        this.worker.rhoasService = this.rhoasService;
-        this.worker.connectorsApi = this.connectorsApi;
-        this.worker.workManager = this.workManager;
-        this.worker.maxRetries = 3;
-        this.worker.timeoutSeconds = 60;
+    public void setup() {
+        databaseManagerUtils.cleanUp();
```

Comment:
@manstis I think if we instead use here the `DatabaseManagerUtils.cleanUpAndInitWithDefaultShard` then that will solve the foreign key constraints you talk about in the comments WRT the `shardId`.